pkg/cmd/server/start/start_master.go

```diff
@@ -242,10 +242,10 @@ func (o MasterOptions) RunMaster() error {
 			return err
 		}
 
-		if err := os.MkdirAll(path.Dir(o.MasterArgs.GetConfigFileToWrite()), os.FileMode(0755)); err != nil {
```

Comment:
Yeah, without the execute bit you cannot read the directory entry and therefore the contents of any file in that directory. While I'm mainly concerned with master-config.yaml there may be other files placed in the same directory at a later date that may contain sensitive data.

```
[root]# mkdir /test
[root]# echo "hi" /test/file
hi /test/file
[root]# echo "hi" > /test/file
[root]# sudo -u bob cat /test/file
hi
[root]# chmod o-x /test
[root]# sudo -u bob cat /test/file
cat: /test/file: Permission denied
```